xkb.c

```diff
@@ -297,13 +297,29 @@ xkb_reload_keymap(void)
     }
 }
 
+void
+xkb_refresh(void)
+{
+    lua_State *L = globalconf_get_lua_State();
+
+    if (globalconf.xkb_reload_keymap)
+        xkb_reload_keymap();
+    if (globalconf.xkb_map_changed)
+          signal_object_emit(L, &global_signals, "xkb::map_changed", 0);
+    if (globalconf.xkb_group_changed)
+          signal_object_emit(L, &global_signals, "xkb::group_changed", 0);
+
+    globalconf.xkb_reload_keymap = false;
+    globalconf.xkb_map_changed = false;
+    globalconf.xkb_group_changed = false;
```

Comment:
> This would be faster because you could do if (!xkb_reload_component) return;

Code optimisation is GCC's job. Also, a cache miss will fetch the whole cache line and there is a high probability that these three booleans will end up in the same cache line. So, sorry, but I do not let "faster" count as a reason here.

> But as you would say, this is bikeshedding. 

Thanks anyway. Sorry for disagreeing with so many of your optimisations...

Also: Wow, you are not complaining about breaking some undocumented API? (One of the signal loses its argument)